fix: バックエンドのCIがpnpm/action-setupの部分で落ちる

### 問題

`pnpm/action-setup`が正常に実行できず、`Build & Check (backend)`が失敗する
e.g. https://github.com/poporonnet/kcmsx/actions/runs/9800457618/job/27062350550#step:4
```
Run pnpm/action-setup@v2
  with:
    version: 9
    dest: ~/setup-pnpm
    run_install: null
    package_json_file: package.json
    standalone: false
Running self-installer...
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 10 seconds. [2](https://github.com/poporonnet/kcmsx/actions/runs/9800457618/job/27062350550#step:4:2) retries left.
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
   ERR_PNPM_META_FETCH_FAIL  GET https://registry.npmjs.org/pnpm: Value of "this" must be of type URLSearchParams
Error: Something went wrong, self-installer exits with code 1
Installation Completed!
```

### 解決

- 同じ現象の報告 https://github.com/pnpm/action-setup/issues/135
  - `pnpm/action-setup`のバージョンが古く、Node16を使用しようとするせい
  - v4なら動くらしい
- そもそも`pnpm/action-setup`を使わず`corepack`にするのが良さそう
  - https://github.com/pnpm/action-setup/issues/99#issuecomment-1918361558
    ```yaml
    - run: corepack enable
    - uses: actions/setup-node@v4
      with:
        node-version: 20
        cache: "pnpm"
    
    - name: Install dependencies
       run: pnpm install
       ```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,19 +11,12 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - run: corepack enable
       - name: Setup Nodejs v20
         uses: actions/setup-node@v4
-
-      - uses: pnpm/action-setup@v2
         with:
-          version: 9
-
-      - name: Cache
-        uses: actions/cache@v4
-        with:
-          path: ~/.cache/install/cache
-          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
-          restore-keys: ${{ runner.os }}-pnpm-
+          node-version: 20
+          cache: "pnpm"
 
       - name: Install packages
         run: pnpm i --frozen-lockfile

--- a/package.json
+++ b/package.json
@@ -19,5 +19,6 @@
   "version": "0.0.1",
   "workspaces": [
     "packages/*"
-  ]
+  ],
+  "packageManager": "pnpm@9.3.0"
 }

